timelock-impl/src/main/java/com/palantir/atlasdb/backup/AtlasBackupResource.java

```diff
@@ -95,11 +118,17 @@ private InProgressBackupToken prepareBackup(Namespace namespace) {
     @Override
     public ListenableFuture<CompleteBackupResponse> completeBackup(
             AuthHeader authHeader, CompleteBackupRequest request) {
-        return handleExceptions(() -> completeBackupInternal(request));
+        return handleExceptions(() -> completeBackupInternal(authHeader, request));
     }
 
     @SuppressWarnings("ConstantConditions")
-    private ListenableFuture<CompleteBackupResponse> completeBackupInternal(CompleteBackupRequest request) {
+    private ListenableFuture<CompleteBackupResponse> completeBackupInternal(
+            AuthHeader authHeader, CompleteBackupRequest request) {
+        if (!permittedBackupToken.get().equals(authHeader.getBearerToken())) {
```

Comment:
This cannot throw NPE right?

Reply:
It shouldn't, but I can be defensive here.